File: source/SharpFlame/ctrlPathSet.cs
using System;
using System.Windows.Forms;

namespace SharpFlame
{
    public partial class ctrlPathSet
    {
        public int SelectedNum
        {
            get { return lstPaths.SelectedIndex; }
            set { lstPaths.SelectedIndex = value; }
        }

        public string SelectedPath
        {
            get
            {
                if ( lstPaths.SelectedIndex < 0 )
                {
                    return null;
                }
                else
                {
                    return (lstPaths.Items[lstPaths.SelectedIndex]).ToString();
                }
            }
        }

        public string[] GetPaths
        {
            get
            {
                string[] Paths = new string[lstPaths.Items.Count];
                int A = 0;
                for ( A = 0; A <= lstPaths.Items.Count - 1; A++ )
                {
                    Paths[A] = (lstPaths.Items[A]).ToString();
                }
                return Paths;
            }
        }

        public ctrlPathSet(string Title)
        {
            InitializeComponent();

            gbxTitle.Text = Title;
        }

        public void SetPaths(modLists.SimpleList<string> NewPaths)
        {
            int A = 0;

            lstPaths.Items.Clear();
            for ( A = 0; A <= NewPaths.Count - 1; A++ )
            {
                lstPaths.Items.Add(NewPaths[A]);
            }
        }

        public void btnAdd_Click(Object sender, EventArgs e)
        {
            FolderBrowserDialog DirSelect = new FolderBrowserDialog();

            if ( lstPaths.Items.Count > 0 )
            {
                DirSelect.SelectedPath = Convert.ToString(lstPaths.Items[lstPaths.Items.Count - 1]);
            }

            if ( DirSelect.ShowDialog() != DialogResult.OK )
            {
                return;
            }

            lstPaths.Items.Add(DirSelect.SelectedPath);
            lstPaths.SelectedIndex = lstPaths.Items.Count - 1;
        }

        public void btnRemove_Click(Object sender, EventArgs e)
        {
            if ( lstPaths.SelectedIndex < 0 )
            {
                return;
            }

            lstPaths.Items.RemoveAt(lstPaths.SelectedIndex);
        }
    }
}

File: reference/FlaMEVB/ctrlPathSet.vb

Public Class ctrlPathSet

    Public Property SelectedNum As Integer
        Get
            Return lstPaths.SelectedIndex
        End Get
        Set(value As Integer)
            lstPaths.SelectedIndex = value
        End Set
    End Property

    Public ReadOnly Property SelectedPath As String
        Get
            If lstPaths.SelectedIndex < 0 Then
                Return Nothing
            Else
                Return CStr(lstPaths.Items.Item(lstPaths.SelectedIndex))
            End If
        End Get
    End Property

    Public ReadOnly Property GetPaths As String()
        Get
            Dim Paths(lstPaths.Items.Count - 1) As String
            Dim A As Integer
            For A = 0 To lstPaths.Items.Count - 1
                Paths(A) = CStr(lstPaths.Items.Item(A))
            Next
            Return Paths
        End Get
    End Property

    Public Sub New(Title As String)
        InitializeComponent()

        gbxTitle.Text = Title
    End Sub

    Public Sub SetPaths(NewPaths As SimpleList(Of String))
        Dim A As Integer

        lstPaths.Items.Clear()
        For A = 0 To NewPaths.Count - 1
            lstPaths.Items.Add(NewPaths(A))
        Next
    End Sub

    Private Sub btnAdd_Click(sender As System.Object, e As System.EventArgs) Handles btnAdd.Click
        Dim DirSelect As New Windows.Forms.FolderBrowserDialog

        If lstPaths.Items.Count > 0 Then
            DirSelect.SelectedPath = CType(lstPaths.Items(lstPaths.Items.Count - 1), String)
        End If

        If DirSelect.ShowDialog <> DialogResult.OK Then
            Exit Sub
        End If

        lstPaths.Items.Add(CStr(DirSelect.SelectedPath))
        lstPaths.SelectedIndex = lstPaths.Items.Count - 1
    End Sub

    Private Sub btnRemove_Click(sender As System.Object, e As System.EventArgs) Handles btnRemove.Click

        If lstPaths.SelectedIndex < 0 Then
            Exit Sub
        End If

        lstPaths.Items.RemoveAt(lstPaths.SelectedIndex)
    End Sub
End Class
